### PR TITLE
Adjust banner for $ProjectRevision

### PR DESCRIPTION
**Pull request by aluan-r7**
## Description
Updated $ProjectRevision banner so that it description is changed to Treck TCP/IP instead of HP printers.
Added examples of other version formats found and added example from running the Treck demo. Treck demo HTTP server included extra words Last Checkpoint e.g. $Project Revision: Last Checkpoint: 4.2.2.13
Changed regex to capture all versions.
added service and software param name values and added capture group for version

## Motivation and Context
Changes were made to fingerprint Treck TCP/IP software running on devices in order to run vulnerability checks.

## How Has This Been Tested?
A clear and concise description of your changes were tested.
Rake test passed
Screen Shot 2020-07-13 at 3 19 25 PM

## Checklist:

- [x] I have updated the documentation accordingly (or changes are not required).
- [x] I have added tests to cover my changes (or new tests are not required).
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by cshannon-r7** [comment]
NIT: Can you add the DCA number to the start of your PR 
DCA-20352: I am a title

File: xml/http_servers.xml [hunk 1]
```diff
@@ -2542,12 +2542,15 @@
...
+  <fingerprint pattern="^\$ProjectRevision:[\s\w:]* ([\d\.]+) \$$">
+    <description>This banner is used to see if devices have Treck TCP/IP</description>
     <example>$ProjectRevision: 4.0.2.38 $</example>
-    <param pos="0" name="os.vendor" value="HP"/>
-    <param pos="0" name="os.device" value="Printer"/>
-    <param pos="0" name="os.family" value="LaserJet"/>
+    <example>$ProjectRevision: 4.2 $</example>
+    <example>$ProjectRevision: 6.0.1.5 $</example>
+    <example>$ProjectRevision: Last Checkpoint: 4.2.2.13 $</example>
+    <param pos="0" name="service.vendor" value="Treck"/>
```
Comment (rkirk-r7) [inline xml/http_servers.xml:2551]: @tsellers-r7 what is your view on this fingerprint, is service appropriate or would service.component make more sense? Or maybe software?

Comment (tsellers-r7) [reply xml/http_servers.xml:2551]: Based on a quick refresher I'd say that while this looks odd, it appears to be the `httpd` bundled with the Treck TCP/IP stack. It also aligns with how NIST has assigned the CPE for Treck so a CPE can be assigned to this fingerprint by our automation.

https://nvd.nist.gov/products/cpe/search/results?keyword=Treck&status=FINAL&orderBy=CPEURI&namingFormat=2.3

**Comment by tsellers-r7** [comment]
@aluan-r7  - Can you run the CPE code on this fingerprint and update the PR?

```
python3 -m venv venv
source venv/{bin,Scripts}/activate
pip install -r requirements.txt

curl -o official-cpe-dictionary_v2.3.xml.gz https://nvd.nist.gov/feeds/xml/cpe/dictionary/official-cpe-dictionary_v2.3.xml.gz && \
gunzip official-cpe-dictionary_v2.3.xml.gz

./update_cpes.py xml/http_servers.xml official-cpe-dictionary_v2.3.xml cpe-remap.yaml && ruby bin/recog_cleanup
```

**Comment by aluan-r7** [comment]
@tsellers-r7 i did it but nothing really seemed to have changed?

**Comment by tsellers-r7** [comment]
@aluan-r7 - I found the issue, will push a change up in a bit.

**Comment by aluan-r7** [comment]
@tsellers-r7 waiting on a pr to add tags to vulnerability content, should i merge this after everything is ready or now?

**Comment by tsellers-r7** [comment]
Are you adding something to Recog or is this product only changes?

**Comment by aluan-r7** [comment]
Theres no more changes for recog other than this

**Comment by tsellers-r7** [comment]
Ok, I can land it now if you like, address a few things, and release a gem.

**Comment by aluan-r7** [comment]
Go ahead

**Pull request merged**